node/chainSimulator/components/testOnlyProcessingNode.go

```diff
@@ -335,6 +355,10 @@ func (node *testOnlyProcessingNode) collectClosableComponents() {
 	node.closeHandler.AddComponent(node.NetworkComponentsHolder)
 	node.closeHandler.AddComponent(node.StatusCoreComponents)
 	node.closeHandler.AddComponent(node.CoreComponentsHolder)
+	node.closeHandler.AddComponent(node.facadeHandler)
+	if enableHTTPServer {
```

Comment:
instead of this ugly if, the createHttpServer can return a mock http server if the chain simulator is run without APIs

Reply:
removed